JavaScript/JavaScript_Limitations.md

```diff
@@ -0,0 +1,85 @@
+
+Web pages hosted on different domains cannot be accessed using JavaScript.  This is based on the "Same origin policy", which prevents a document or script loaded from one origin, from getting or setting properties of a document from another origin. This policy dates all the way back to Netscape Navigator 2.0.  It helps to ensure that potentially malicious documents/scripts can be isolated.
+
+A workaround is in place with HTML5 using postMessage interface, or using server side scripts using AJAX calls.  There are a few methods to do this the common ones are `document.domain` method which is an iframe method that sets the value of document.domain to a suffix of the current domain.
+````
+For example, assume a script in the document at `http://shop.company.com/dir/other.html`
+which executes the following statement:
+
+document.domain = "company.com";
+
+After execution, the page would pass the origin check with http://company.com/dir/page.html.
+However, by the same reasoning, company.com could not set document.domain to othercompany.com.
+
+````
+Using this you can execute javascript from an iframe.  Please note that this is not suitable for Firefox which will not allow a change to `document.domain`
+
+Another notable method is `window.postMessage` which safely enables cross-origin communication.  Cross-window messaging security model is two-sided. The sender ensures that the receiving domain is targetDomain. The receiver checks that the message came from proper event.origin.
+
+````
+postMessage(data, targetDomain), where:
+
+data - The message. According to the specification, it could be any object. But as of now, only strings are supported in major browsers.
+
+targetDomain - Limit receiving iframe by given domain. Can contain ‘*’ which doesn’t put any restrictions.
+````
+
+
+#### Database Access
+
+Accessing databases, in particular backend databases, is a problem without the help of a server side script like Ajax which performs the database access.  
+
+For example if a user was to select an item in a dropdown list, which is implemented by JavaScript and these items are in a database.  A function could be used to show the dropdown details.  This function would do the following.
+1. Check if item is selected.
+2. Create an XMLHttpRequest object.
+3. Create the function to be executed when the server response is ready.
+4. Send the request off to a file on the server.
+
+Then in turn the page on the server called by the JavaScript above is an ASP file called "getdetails.asp".  This file would run a query against a database, and return the result in an HTML table
+
+
+#### Writing files to the server
+
+JavaScript cannot write to files on the server without the help of a server side script. Using Ajax, JavaScript can send a request to the server, which can read a file in XML or plain text format but it cannot write to a file unless the file called on the server actually runs as a script to do the file write for you.
```

Comment:
Very long sentence. Can split, e.g.:
... or plain text format. However, it cannot...